tsdb/exemplar_test.go

```diff
@@ -499,3 +500,40 @@ func BenchmarkResizeExemplars(b *testing.B) {
 		})
 	}
 }
+
+func TestCircularExemplarStorage_Concurrent_AddExemplar_Resize(t *testing.T) {
+	exs, err := NewCircularExemplarStorage(0, eMetrics)
+	require.NoError(t, err)
+	es := exs.(*CircularExemplarStorage)
+
+	l := labels.FromStrings("service", "asdf")
+	e := exemplar.Exemplar{
+		Labels: labels.FromStrings("trace_id", "qwerty"),
+		Value:  0.1,
+		Ts:     1,
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	t.Cleanup(wg.Wait)
+
+	started := make(chan struct{})
+
+	// Try to provoke data race between AddExemplar and Resize.
```

Comment:
nit: maybe we could move this to the test's doc comment?

Reply:
Done, PTAL.